src/utils/getTokenPriceInUSD.ts

```diff
@@ -0,0 +1,175 @@
+        { internalType: "uint80", name: "roundId", type: "uint80" },
+        { internalType: "int256", name: "answer", type: "int256" },
+        { internalType: "uint256", name: "startedAt", type: "uint256" },
+        { internalType: "uint256", name: "updatedAt", type: "uint256" },
+        { internalType: "uint80", name: "answeredInRound", type: "uint80" },
+      ],
+      stateMutability: "view",
+      type: "function",
+    },
+    {
+      inputs: [],
+      name: "latestRoundData",
+      outputs: [
+        { internalType: "uint80", name: "roundId", type: "uint80" },
+        { internalType: "int256", name: "answer", type: "int256" },
+        { internalType: "uint256", name: "startedAt", type: "uint256" },
+        { internalType: "uint256", name: "updatedAt", type: "uint256" },
+        { internalType: "uint80", name: "answeredInRound", type: "uint80" },
+      ],
+      stateMutability: "view",
+      type: "function",
+    },
+    {
+      inputs: [],
+      name: "version",
+      outputs: [{ internalType: "uint256", name: "", type: "uint256" }],
+      stateMutability: "view",
+      type: "function",
+    },
+  ];
+  // The address of the contract which will provide the price of ETH
+  const feedAddress = tokenAddressToFeedAddress(chainId, tokenAddress);
+
+  if (!feedAddress) {
+    throw new Error("Token not supported");
+  }
+
+  // We create an instance of the contract which we can interact with
+  const priceFeed = new ethers.Contract(
+    feedAddress,
+    aggregatorV3InterfaceABI,
+    provider,
+  );
+  // We get the data from the last round of the contract
+  // Determine how many decimals the price feed has (10**decimals)
+  const [roundData, decimals] = await Promise.all([
+    priceFeed.latestRoundData(),
+    priceFeed.decimals(),
+  ]);
+  // We convert the price to a number and return it
+  return Number(
+    (roundData.answer.toString() / Math.pow(10, Number(decimals))).toFixed(2),
+  );
+};
+
+const tokenAddressToFeedAddress = (chainId: ChainId, tokenAddress: string) => {
+  const currencies = currenciesByNetwork[chainId];
+  const currency = Object.values(currencies).find(
+    (currency) => currency.address === tokenAddress,
+  );
+
+  if (!currency) {
+    throw new Error("Currency not supported");
+  }
+
+  const symbol = currency.symbol as keyof (typeof currenciesByNetwork)[ChainId];
+  const feedsForChain = feedsPerChain[chainId];
+
+  if (!feedsForChain) {
+    throw new Error("Chain not supported");
+  }
+
+  return feedsForChain?.[symbol];
+};
+
+export const getTokenPricesForChain = async (chainId: ChainId) => {
+  const currencies = currenciesByNetwork[chainId];
+  const prices = await Promise.all(
+    Object.values(currencies).map(async (currency: Currency) => {
+      try {
+        const price = await getTokenPriceInUSD(chainId, currency.address);
+        return { ...currency, price };
+      } catch (error) {
+        console.error(error);
+        return { ...currency, price: 0 };
```

Comment:
What do you think is the best solution here? Should we throw an error and simply not return the cheapest hypercert if we cannot fetch the current price for a specific currency? Or not consider orders for being the cheapest if we cannot determine the current token price for the currency of that order?

This would currently only be used for determining which is the cheapest, but I imagine we will need these utilities in the frontend as well, so it might be useful to think about this for a second.

Reply:
I'm just throwing an error for now, which should be caught in the top level. If another solution is preferred, please let me know.

Reply:
> This would currently only be used for determining which is the cheapest, but I imagine we will need these utilities in the frontend as well, so it might be useful to think about this for a second.

This data field `cheapestOrder` might not even belong in the Graph, but let's leave it there for now so we can ship